app/gui2/ydoc-server/fileFormat.ts

```diff
@@ -0,0 +1,115 @@
+import * as json from 'lib0/json'
+import z from 'zod'
```

Comment:
very cool. we may want to consider using typebox, if we're fine with generating JS at runtime:
https://moltar.github.io/typescript-runtime-type-benchmarks/